components/FormikFields/PaperInputPicker/index.js

```diff
@@ -34,6 +37,70 @@ const PaperInputPicker = ({
 
   const [location, setLocation] = React.useState({ latitude: 0, longitude: 0, altitude: 0 });
   const [locationLoading, setLocationLoading] = React.useState(false);
+  const [numberedQestions, setNumberedQuestions] = React.useState({});
+  const [questionsToRepeat, setQuestionsToRepeat] = React.useState([]);
+  const [additionalQuestions, setAdditionalQuestions] = React.useState([]);
+  const [individualLoopsAdded, setIndividualLoopsAdded] = React.useState(0);
+
+  useEffect(() => {
+    if (fieldType === 'loop') {
+      numberQuestionsInConfig();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (fieldType === 'loop') {
+      updateQuestionsToRepeat();
+    }
+  }, [numberedQestions]);
+
+  // number all questions in config (for looping ccapabilities)
+  const numberQuestionsInConfig = () => {
+    const fieldsNumberedJson = {};
+    config.fields.forEach((field, index) => {
+      fieldsNumberedJson[index] = field;
+    });
+    setNumberedQuestions(fieldsNumberedJson);
+  };
+
+  // update the questions that will be repeated if the field type is loop
+  const updateQuestionsToRepeat = () => {
+    Object.entries(numberedQestions).forEach(([key, value]) => {
+      if (value.formikKey === formikKey) {
+        let repeatQuestions = [];
+        for (let i = key - numberQuestionsToRepeat; i < key; i += 1) {
+          // ensure question does not excceed the first question
+          if (i >= 0) {
+            repeatQuestions = repeatQuestions.concat(numberedQestions[i]);
+          }
+        }
+        setQuestionsToRepeat(repeatQuestions);
+      }
+    });
+  };
+
+  // add another loop of questions in correct position in form (additionalQuestions)
+  // update number of loops added (+1)
+  const addLoop = () => {
+    // updated formikKey to ensure uniqueness
+    let updatedQuestions = [];
+    questionsToRepeat.forEach((question) => {
+      const updatedQuestion = _.cloneDeep(question);
+      updatedQuestion.formikKey = `${updatedQuestion.formikKey}__loop${individualLoopsAdded}`;
+      updatedQuestions = updatedQuestions.concat(updatedQuestion);
+    });
+
+    setAdditionalQuestions(additionalQuestions.concat(updatedQuestions));
+    setLoopsAdded(loopsAdded + 1);
+    setIndividualLoopsAdded(individualLoopsAdded + 1);
+  };
+
+  const removeLoop = () => {
+    setLoopsAdded(loopsAdded - 1);
+    setIndividualLoopsAdded(individualLoopsAdded - 1);
+    setAdditionalQuestions(additionalQuestions.slice(0,
+      additionalQuestions.length - numberQuestionsToRepeat));
+  };
```

Comment:
@jamccomb92 so my only hesitance is that this is a lot of loop state management to add to the `<PaperInputPIcker>` it might be a good opportunity to start creating a `loop.js` with a `<Looper />` component that we can import and move here

I can do it or at least walk you through it